src/main/scala/com.socrata.tileserver/services/TileService.scala

```diff
@@ -227,13 +228,40 @@ object TileService {
     val dis = rs.open(new DataInputStream(resp.inputStream(Long.MaxValue)))
 
     try {
-      val headers = MsgPack.unpack(dis, MsgPack.UNPACK_RAW_AS_STRING).asInstanceOf[Map[String, Any]]
-      val jsonHeaders = JObject(headers.mapValues(v => JString(v.toString)))
-      headers.asInt("geometry_index") match {
-        case geomIndex: Any if geomIndex < 0 =>
-          Failure(InvalidSoqlPackException(headers))
-        case geomIndex: Any =>
-          Success(jsonHeaders -> new FeatureJsonIterator(dis, geomIndex))
+      val soqlIter = new SoQLPackIterator(dis)
+      val jsonHeaders = JObject(soqlIter.headers.mapValues(v => JString(v.toString)))
+      if (soqlIter.geomIndex < 0) {
+        Failure(InvalidSoqlPackException(soqlIter.headers))
+      } else {
+        val colNames = soqlIter.schema.map(_._1).toArray
+        val featureJsonIter = soqlIter.map { case soqlRow =>
```

Comment:
Yeah, not sure how to apply it to multiple things.  I think what can be done is to have a base kind of SoQLPackIterator which can map to different values, but will see.